tests/test_cleanname.py

```diff
@@ -86,3 +86,8 @@ def test_with_unicode_umlauted_name(terms):
    errmsg = "preserving cleanup of %s failed"
    for testname, (variation, expected) in unicode_umlaut_tests.items():
       assert basename(variation, terms, prefix=True) == expected, errmsg % testname
+
+
+def test_deterministic_cleanup_for_two_types(terms):
```

Comment:
I find this test hard to comprehend. How do these two assertions prove deterministic operation?

I am also surprised that `basename` strips away both suffixes. I don't think it should do that. But that issue seems to exist on current code too.

Reply:
This test will pass only if the terms are ordered such that "inc." is before "ltd".
The version in the master does not guarantee such order, and depending on the initialization of python variables (PYTHONHASH) in some runs would get this ordering and some will not. So in case of non-deterministic sorting this test will fail with some probability.
I know that this is not perfect solution, but I'm not able to find anything better (you are not able to change PYTHONHASH after initialization).

Reply:
And about removing both terms, actually it happens a lot (at least in my data) so this is useful behaviour in current code.

Reply:
> This test will pass only if the terms are ordered such that "inc." is before "ltd".

> And about removing both terms, actually it happens a lot (at least in my data) so this is useful behaviour in current code.

Thanks. I understand the problem and the solution and it's fine by me. And I agree that being able to remove multiple terms is useful. 

But taking a took a look at the code again, I remember that the multiple-terms removal is totally non-deterministic. I actually removed a `multi=True|False` option from `basename` because it is not working deterministically either way. 

Hence I find it problematic that the test is based on this non-deterministic behavior (that should and hopefully will be fixed some day).

Reply:
I understand your point and I full agree. I've changed the tests to actually test whatever the prepare_terms depends on the ordering of terms or not.